lib/spack/spack/package.py

```diff
@@ -689,6 +691,23 @@ def global_license_file(self):
         return os.path.join(self.global_license_dir, self.name,
                             os.path.basename(self.license_files[0]))
 
+    @property
+    def unique_id(self):
```

Comment:
not sure why this should go in `package.py`, as this identifier is really for parallel builds.  There are lots of granularities we can use (dag hash, build hash, full hash, name/version, etc.) but the granularity depends on the purpose so I would put this in `installer.py` (the purpose), not here.

Reply:
The `unique_id` is in a sense a property of the package that I originally intended to use the name, version, and dag hash.  I only removed the hash when I had problems resolving install issues where packages had dependencies with conflicting versions.

Are you suggesting a function in the `installer` to "calculate" the id for each package it encounters?

Reply:
I’m saying there are lots of ids packages provide (hashes,  Ames, etc)  but this one’s the one we chose (for now) for the installer, and what criteria you use for uniqueness depends on the application. e.g., for installations we currently use DAG hashes but we will eventually change to full hashes. For single-DAG operations and in packages we just use names. When reading installer, I found myself wondering what the id was, and only found it at the bottom of the PR, so I think this function should just be a free function in installer — especially since all the comments here are things you’d want to know when looking at installer.

Reply:
Ah.  That's an excellent point.  Thanks for clarifying.

Reply:
Moved to installer and renamed to `package_id` (since not currently unique).